generic_config_updater/change_applier.py

```diff
@@ -160,18 +162,33 @@ def apply(self, change):
             log_error("Failed to apply Json change")
         return ret
 
-
     def remove_backend_tables_from_config(self, data):
         for key in self.backend_tables:
             data.pop(key, None)
 
-
     def _get_running_config(self):
-        (_, fname) = tempfile.mkstemp(suffix="_changeApplier")
-        os.system("sonic-cfggen -d --print-data > {}".format(fname))
-        run_data = {}
-        with open(fname, "r") as s:
-            run_data = json.load(s)
-        if os.path.isfile(fname):
+        _, fname = tempfile.mkstemp(suffix="_changeApplier")
+        print(fname)
+        
+        if self.namespace:
+            cmd = ['sonic-cfggen', '-d', '--print-data', '-n', self.namespace]
+        else:
+            cmd = ['sonic-cfggen', '-d', '--print-data']
+
+        with open(fname, "w") as file:
+            result = subprocess.Popen(cmd, stdout=file, stderr=subprocess.PIPE, text=True)
```

Comment:
>file [](http://example.com/codeflow?start=50&length=4)

We could optimize and remove this tempfile.

Reply:
This will be optimized in next PR to refactor the UT organization part.